### PR TITLE
Storybook: Extend guidance in README 

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
In order to move further with storybook, this PR sets to define the expected goal in more detail. 
This is a suggestion and should facilitate a discussion for best practice. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: storybook/README.md [hunk 1]
```diff
@@ -1,17 +1,56 @@
...
+The API section contains stories for all components. The goal is to communicate a components behaviour as concise and simple as possible.
 
-- Additionally we will want to add stories that document complex usage of recharts with custom components.
+`storybook/stories/API/cartesian/Line.stories.tsx` can be considered a template for writing stories for the API section.
 
-Best practice: 
-- A story for a LineChart should not contain a complex brush. Each story should focus on a single component and its behaviour. If the interaction with the brush is to be shown, then any other brush details should ideally be omitted. 
-- Complex interaction effect between multiple component best deserve a separate story. 
+Acceptance criteria for API stories:
+- Each component should be documented in a doc file / mdx file.
```
Comment (ckifer) [inline storybook/README.md:17]: nit: can we just stick to not using `.mdx` to be consistent?

File: storybook/README.md [hunk 2]
```diff
@@ -1,17 +1,56 @@
...
+- All stories for a single component are found in a single file.
```
Comment (ckifer) [inline storybook/README.md:18]: What do we mean by "single component"? "component" is very general in the context of react

**Review by ckifer — APPROVED** [review]

**Comment by ckifer** [comment]
LGTM, one nit and one clarification

File: storybook/README.md (hunk 1 above)
Comment (nikolasrieble) [reply storybook/README.md:17]: It so turns out that for proper documentation we do need mdx. For an example I started here https://github.com/recharts/recharts/pull/3404

File: storybook/README.md (hunk 2 above)
Comment (nikolasrieble) [reply storybook/README.md:18]: Good point. I would argue every single recharts component that is exported. 🤔

**Comment by nikolasrieble** [comment]
> LGTM, one nit and one clarification

I will wait a couple of days to get some input from @proke03 and @Yilun-Sun.

**Review by proke03 — COMMENTED** [review]
Sorry for the late Comment.
Looks good to me.
In particular, the explanation of examples is clear.
Thank you!

File: storybook/README.md [hunk 3]
```diff
@@ -1,17 +1,56 @@
...
 
-- We want to document every single component in
--- the most simple manner
--- a complex usage
--- edge cases
+### API
+The API section contains stories for all components. The goal is to communicate a components behaviour as concise and simple as possible.
 
-- Additionally we will want to add stories that document complex usage of recharts with custom components.
+`storybook/stories/API/cartesian/Line.stories.tsx` can be considered a template for writing stories for the API section.
```
Comment (peterklingelhofer) [inline storybook/README.md:14]: We could link them to the file on GitHub here as well if we wanted (super minor suggestion definitely feel free to ignore):

```suggestion
[`storybook/stories/API/cartesian/Line.stories.tsx`](https://github.com/recharts/recharts/blob/master/storybook/stories/API/cartesian/Line.stories.tsx) can be considered a template for writing stories for the API section.
```

Comment (nikolasrieble) [reply storybook/README.md:14]: Great idea. All suggestions are very welcome!

**Review by peterklingelhofer — APPROVED** [review]

**Pull request merged**